### ISSUE TITLE
Add documentation on how to contribute

### ISSUE THREAD

**Issue by SimplyComple0x78** [HelloVolla/android-launcher-qt#31]
Hey there,

I want to get involved into the project and miss information on how to do so:
* **What IDE** to use (that's a big question I have currently, not having Qt Creator and I think I can't get it for free, right?),
* how to **open issues** (e.g. currently there is no issue for missing messenger-support like WhatsApp - shall I open one?),
* whether there are any **automated tests** that need to pass,
* the overall flow (who may or may not open pull-requests for open issues, who reviews them?),
* etc.

Of course, these are specific questions I personally ask myself, but from a community point-of-view there are questions others may ask, too. Thus someone (who can answer these questions) should add some documentation for it.

Nevertheless, I like to get my very own answers soon, if possible.

Best wishes

**Comment by nift4** [comment]
Hey, technically, you are not allowed to by the license.

**Comment by wurzer** [comment]
You are welcome to contribute. 

1. I recommend to use the community edition of the Qt Creator.
2. You can open issues right here in Guthub, that we use next to our internal board.
3. We have currently automatic tests after you have created a pull request. However we should introduce unit tests.
4. The review of pull requests is currently done by internal team members.

Regarding the license: the license allow to modify the code for your own, but not distribution and commercialization. Thanks for the motivating inquiry.

**Comment by nift4** [comment]
Hello, the license does not allow redistrubution of the source or changed versions. Is that still up to date?

**Comment by wurzer** [comment]
Yes, you can copy, modify and use a version for your self but the license excludes a redistribution. However you could create and maintain a branch right here at Github. And the Volla team would appreciate your contribution to the master branch. I have send you an invitation to join the team.

**Comment by nift4** [comment]
I see, thanks!

**Comment by 12people** [comment]
@wurzer The wording of the license here is a bit confusing:
"use, copy, modify, and/or merge in the Software for personal,
non-commercial use, but not to publish, distribute, sublicense, and/or sell the Software or
a copy of the Software, even in a part"

Creating a public Github repository with the same code sounds like publishing the software to me, no?

(Either way, it's worth noting that whether it does or does not, the license is proprietary and does not benefit the free software community at all.)

**Comment by wurzer** [comment]
You are right. However you can contribute with a branch or use a private repository. 

Unfortunately we didn't find a license, that matches our requirements. However we will periodically reflect the license and may publish the code unser an updated or difference license. 

Developers can modify, build and use the code for free for personal use.
